testsuite/tests/apicast/operator/conftest.py

```diff
@@ -0,0 +1,181 @@
+    end
+
+    function _M:log()
+    -- can do extra logging
+    end
+
+    function _M:balancer()
+    -- use for example require('resty.balancer.round_robin').call to do load balancing
+    end
+
+    return _M
+    """
+    policydict = {
+        "init.lua": initlua,
+        "example.lua": examplelua,
+        "apicast-policy.json": apicastpolicyjson
+    }
+
+    secrets.create(name="policy", string_data=policydict)
+    return secrets
+
+
+# pylint: disable=unused-argument
+@pytest.fixture(scope="module")
+def patch_apimanager(request, openshift, create_custom_policy_secret):
+    """Patch the apimanager CRD to add the custom policy added to apicast staging
+    """
+    apimanager = openshift().api_manager
+    # TODO: do it through apimanager.set_path but it looks not working at this time
+    openshift().patch("APIManager", apimanager.name(),
+                      patch={"spec":
+                             {"apicast":
+                              {"stagingSpec":
+                               {"customPolicies": [
+                                {"name":
+                                    "example",
+                                    "version": "0.1",
+                                    "secretRef":
+                                    {"name": "policy"}
+                                 }],
+                                "replicas": 1
+                                }}}}, patch_type="merge")
+    request.addfinalizer(lambda: openshift().patch("APIManager", apimanager.name(),
+                         patch={"spec":
+                                {"apicast": {
+                                    "stagingSpec": {
+                                        "customPolicies": []
+                                    }
+                                }}}, patch_type="merge"))
+
+    @backoff.on_predicate(backoff.fibo, max_tries=10)
+    def wait_for_starting(apimanager):
+        """Check that apicast-staging is not in ready state
+        """
+        return not apimanager.ready({"apicast-staging"})
+
+    @backoff.on_predicate(backoff.fibo, max_tries=10)
+    def wait_for_ready(apimanager):
+        """Waiting for both apicast being in readdy state
+           to ensure that the whole apimanager is up and running
+        """
+        return apimanager.ready({"apicast-staging", "apicast-production"})
+
+    # We need to explicitly wait for the deployment being in starting state
+    # before waiting for it to be ready, the operator needs time to reconcile
+    # its state
+    wait_for_starting(apimanager)
+    wait_for_ready(apimanager)
+
+
+@pytest.fixture(scope="module")
+def service_settings(request) -> dict:
+    "dict of service settings to be used when service created"
+    return {"name": blame(request, "svc")}
+
+
+# pylint: disable=too-many-arguments, too-many-instance-attributes
+@pytest.fixture(scope="module")
+def service(backends_mapping, custom_service, service_settings,
```

Comment:
maybe this can be done easier. as it is in testsuite/tests/apicast/policy/conftest.py
Actually it may make sense to move operator/ under testsuite/tests/apicast/policy/. thiughts?

Reply:
I agree